realm/realm-library/src/main/java/io/realm/RealmBaseRecyclerAdapter.java

```diff
@@ -0,0 +1,195 @@
+package io.realm;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+
+/**
+ * The RealmBaseRecyclerAdapter class is an abstract utility class for binding RecyclerView UI elements to Realm data.
+ * <p>
+ * This adapter will automatically handle any updates to its data and call {@link #notifyDataSetChanged()} as
+ * appropriate.
+ * <p>
+ * The RealmAdapter will stop receiving updates if the Realm instance providing the {@link OrderedRealmCollection} is
+ * closed.
+ *
```

Comment:
Please also describe current limitations, like `itemRange` callbacks not being supported yet. They will be implemented as part of #989